File: tests/journal.lib/chronicler_sanity.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// alias
using chronicler_t = pyre::journal::chronicler_t;


// verify that the initial chronicler global state is as expected
int main() {
    // verify that the default detail is at level 1
    assert (chronicler_t::detail() == 1);

    // get the global metadata map
    chronicler_t::notes_type & notes = chronicler_t::notes();

    // there should be only one setting for now
    assert (notes.size() == 1);
    // verify the known contents
    assert (notes.at("application") == "journal");

    // get the default device
    auto device_ptr = chronicler_t::device();
    // by default, it is the console; verify that it is not an empty pointer
    assert(device_ptr);
    // and that it's the console
    assert ((*device_ptr).name() == "cout");

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/chronicler_sanity.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


# the journal global state
def test():
    """
    Verify the default global settings
    """
    # get the console
    from journal.Console import Console as cout
    # and the chronicler
    from journal.Chronicler import Chronicler as chronicler

    # ask it for its metadata
    meta = chronicler().notes
    # verify that the table comes with only one setting
    assert len(meta) == 1
    # that is the default application name
    assert meta["application"] == "journal"

    # get the default device
    device = chronicler().device
    # verify it's an instance of {cout}
    assert isinstance(device, cout)
    # and that it is named correctly
    assert device.name == "cout"

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
